tools/mo/openvino/tools/mo/front/tf/loader.py

```diff
@@ -236,7 +244,21 @@ def load_tf_graph_def(graph_file_name: str = "", is_binary: bool = True, checkpo
                 graph_def = frozen_func.graph.as_graph_def(add_shapes=True)
                 # disable eager execution since next steps are executed with a graph in non-eager mode
                 tf_v1.disable_eager_execution()
-                return graph_def, variables_values, 'tf2'
+
+                input_names = []
+                if hasattr(imported, 'inputs'):
+                    # Extract tensor names order from Keras model
+                    input_names = [tensor.name for tensor in imported.inputs]
```

Comment:
why don't we extract input tensor names by using `frozen_func.inputs`. It should also have the same order of inputs.
So you don' t need new code above for TF for loading

Reply:
No, order is not the same. Inputs/outputs in frozen_func are sorted by alphabet, which is not correspond original model. But during model inference the order should be as in original model, not sorted. I checked it in on layer tests. 
frozen_func does not have original order, the order is stored only in saved_model format for TF2. But the only problem is that tensor names of outputs are renamed after freezing (input names are not renamed). For this reason I used frozen_func.outputs which are sorted by "Identity" names.